_rules/SC3-1-2-lang.md

```diff
@@ -1,79 +1,78 @@
 ---
-rule_id: SC3-1-2-lang
-name: Languages within the body
-test_mode: automatic
-environment: DOM Structure
+name: Validity of lang attribute value in html body elements
+
+description:
+- This rule checks that `lang` and/or `xml:lang` attribute on elements within the `body` of a web page conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
 
 success_criterion:
-- 3.1.2 # Language of Parts (Level AA)
+- 3.1.2
+
+test_aspects:
+- DOM Tree
 
 authors:
-- Annika Nietzio
+- Bryn Anderson
+- Jey Nandakumar
 ---
 
-## Description
-
-This test checks that `lang` attributes within the body of a web page are correct.
-
-## Background
+## Test Procedure
 
-- [H58: Using language attributes to identify changes in the human language](http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140408/H58)
-- [eGovMon test for SC3.1.2](http://wiki.egovmon.no/wiki/SC3.1.2#Element_descendent-or-self::body.5B.40lang.5D_or_descendent-or-self::body.5B.40xml:lang.5D)
-- [BCP 47: Tags for the Identification of Languages](http://www.rfc-editor.org/rfc/bcp/bcp47.txt)
+### Applicability
 
-## Assumptions
+- Any DOM element, other than the root DOM node with a [non-empty][] `lang` and/or `xml:lang` attribute.
 
-- The comparison of language-code does not look for exact matches. Technique H57 states: "Use of the primary code is important for this technique." which means that region subtags can be ignored in the comparison, i.e. "en-GB" is the same as "en".
-- This test checks the `lang` attributes. The `xml:lang` attributes are not taken into account because tests have shown, that `xml:lang` is ignored by screenreaders. (Both Jaws 15 with FF and IE and NVDA with FF go by lang attribute, xml:lang is ignored.) The `xml:lang` attributes are checked by a separate test: [SC3-1-2-xml-lang](SC3-1-2-xml-lang.html).
+**Note**: To retrieve all DOM nodes with a `lang` and or `xml:lang` attribute, other than the root node use -
```

Comment:
"To retrieve all DOM _elements_ with a `lang` and or `xml:lang` attribute, other than the root _element_ use"